src/Data/Text/Internal/Fusion/Common.hs

```diff
@@ -230,8 +283,12 @@ head_empty :: a
 head_empty = streamError "head" "Empty stream"
 {-# NOINLINE head_empty #-}
 
--- | /O(1)/ Returns the first character and remainder of a 'Stream
--- Char', or 'Nothing' if empty.
+-- | /O(1)/ Returns the first character and remainder of a `Stream`
+-- `Char`, or 'Nothing' if empty.
+--
+-- __Properties__
+--
+-- @ `Data.Functor.fmap` `Data.Tuple.fst` . `uncons` . `Data.Text.Internal.Fusion.stream` === `Data.Functor.fmap` `Data.Tuple.fst` . `Data.Text.uncons` @
```

Comment:
Can we have a stronger property here, saying something about both components of the tuple?